### PR TITLE
Allow specifying URL as input for custom sign in logo

### PR DESCRIPTION
**Pull request by MaikuMori**
## Description

I didn't use specific URL verification functions from standard lib because just testing for `https:// | http://` seems sufficient.

## Motivation and Context

Adding custom files in K8S to containers is tedious, I would rather link to logo stored on a CDN.

## How Has This Been Tested?

I've added tests. They seem to run :). I'm actually testing Codespaces for this PR, so bear with me.

## Checklist:

- [X] My change requires a change to the documentation or CHANGELOG.
- [X] I have updated the documentation accordingly.
- [x] I have updated the CHANGELOG accordingly.
- [X] I have created a feature (non-master) branch for my PR.

### CODE REVIEW

File: pkg/app/pagewriter/pagewriter.go
Comment (JoelSpeed) [inline pkg/app/pagewriter/pagewriter.go:57]: Nit
```suggestion
	// If a URL is used, image support depends on the browser.
```

**Review by JoelSpeed — COMMENTED** [review]
This seems like a reasonable patch, thanks!

> I didn't use specific URL verification functions from standard lib because just testing for https:// | http:// seems sufficient.

What kind of thing were you planning to do?

Have you tested this at all? Are there any issues with loading resources from different domains reported by the browser?

Added one documentation nit, if you could fix that and add a changelog I'm reasonably happy with this as is

**Comment by MaikuMori** [comment]
> What kind of thing were you planning to do?

I didn't plan. There are functions like [url.ParseRequestURI](https://pkg.go.dev/net/url#ParseRequestURI) which would verify that it's a valid URI, not just a string with a prefix, but I don't think it's needed here.

> Have you tested this at all? 

Yes. Images don't have cross domain restrictions. [A web page may freely embed cross-origin images, stylesheets, scripts, iframes, and videos.](https://en.wikipedia.org/wiki/Cross-origin_resource_sharing)

Since I'm checking for HTTP/HTTPS prefix, only full URLs are supported. No funny business with relative paths and domains.

>  add a changelog

Will do.

**Review by JoelSpeed — DISMISSED** [review]
I think this is good to go then.

As you say, no funny business with URLs etc, I don't think there's really any point validating this as the user will notice straight away if the image is broken.

@NickMeves any security concerns on your side about allowing this?

File: pkg/app/pagewriter/sign_in_page.go
Comment (NickMeves) [inline pkg/app/pagewriter/sign_in_page.go:108]: @JoelSpeed -- 2 security concerns:

- We shouldn't have `http://` be allowed. If the oauth2-proxy is hosted with HTTPS (which it very well should be!), HTTP would result in mixed content warnings/issues with the HTTPS hosting page.
- (Non Blocking) There's a very small security issue potentially on where this IMG is hosted and the `cookie-domain` -- if non-exact domain style settings are used (e.g. `.yourcompany.com`) the session cookies will leak to this IMG path if they are hosted at the same top level domain (potentially an issue since a lot of people use different service providers under different subdomains that have different security controls/ownership).

**Review by NickMeves — CHANGES_REQUESTED** [review]
Let's not allow HTTP hosting.

(@JoelSpeed feel free to override me though 😄 )

**Comment by MaikuMori** [comment]
I'm totally fine with removing HTTP. The only advantage I can think is if someone hosts stuff in a private network, but even that should be done via HTTPS anyway.

**Comment by MaikuMori** [comment]
I've removed HTTP link support. Let me know if there is anything else to be done to merge this.

@JoelSpeed  @NickMeves

File: pkg/app/pagewriter/sign_in_page.go
Comment (JoelSpeed) [inline pkg/app/pagewriter/sign_in_page.go:110]: Not blocking but a tip for the future, instead of `\"%s\"` you can just do `%q` and it has the same effect :slightly_smiling_face:

**Review by JoelSpeed — APPROVED** [review]
This is fine by me, but I'll let @NickMeves hit the merge button as he raised those earlier security concerns

**Review by NickMeves — APPROVED** [review]
LGTM! Thanks for the security fix!

**Pull request merged**
